client/sprites.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"bytes"
+	"image"
+	"image/color"
+
+	"github.com/faiface/pixel"
+	"github.com/mmogo/mmo/client/assets"
+	"github.com/mmogo/mmo/shared"
+)
+
+func init() {
+	AtlasL = atlasL()
+}
+
+// Sprite is an animated sprite
+type Sprite struct {
+	Picture pixel.Picture
+	Frames  map[shared.Direction]map[shared.Action][]pixel.Rect
+	Sprite  *pixel.Sprite
+	Frame   int     // current frame
+	Speed   float64 // frames per second
+	elapsed float64
+}
+
+func (s *Sprite) Animate(dt float64, facing shared.Direction, action shared.Action) {
+	s.elapsed += dt
+	if s.Speed == 0 {
+		s.Speed = 0.1
+	}
+	s.Frame = 0 // default frame
+	if facing == shared.DIR_NONE {
+		facing = DOWN
+	}
+	if len(s.Frames[facing][action]) > 0 {
+		s.Frame = int(s.elapsed/s.Speed) % len(s.Frames[facing][action])
+	}
+	s.Sprite.Set(s.Picture, s.Frames[facing][action][s.Frame])
+
+}
+
+func (s *Sprite) Draw(target pixel.Target, matrix pixel.Matrix, color color.Color) {
+	if s.Sprite == nil {
+		s.Sprite = pixel.NewSprite(nil, pixel.Rect{})
+	}
+	s.Sprite.DrawColorMask(target, matrix, color)
+}
+
+// Atlas map
+type Atlas map[shared.Direction]map[shared.Action][]pixel.Rect
+
+var AtlasL Atlas
```

Comment:
personaly i like type and var definitions at the top of a file, maybe consider moving this up? (especially to be close by `init()`)